tests/qiskit_convert_test.py

```diff
@@ -760,3 +760,26 @@ def test_qcontrolbox_conversion() -> None:
     tkc2 = qiskit_to_tk(qc2)
     assert tkc2.n_gates == 3
     assert tkc2.n_gates_of_type(OpType.QControlBox) == 3
+
+
+def test_state_prep_conversion() -> None:
+    # State prep with list of real amplitudes
+    ghz_state = [1 / np.sqrt(2), 0, 0, 0, 0, 0, 0, 1 / np.sqrt(2)]
+    qc_sp = QuantumCircuit(3)
+    qc_sp.initialize(ghz_state)
+    tk_sp = qiskit_to_tk(qc_sp)
+    assert tk_sp.n_gates_of_type(OpType.StatePreparationBox) == 1
+    assert tk_sp.n_gates == 1
+    ghz_array = np.array(ghz_state)
+    assert compare_statevectors(tk_sp.get_statevector(), ghz_array)
+    # State prep with ndarray of complex amplitudes
+    qc_sp2 = QuantumCircuit(2)
+    complex_statvector = np.array([0, 1 / np.sqrt(2), -1.0j / np.sqrt(2), 0])
```

Comment:
Please also add a test to verify that we are handling endianness correctlty. (I am not sure which convention qiskit uses here.) Both of the test states used in the tests here are symmetric under reversal of qubits up to global phase (and `compare_statevectors()` only checks up to global phase). For example `[0, 1, 0, 0]` would be a good test.